modules/vpc-sc/variables.tf

```diff
@@ -89,10 +89,11 @@ variable "egress_policies" {
   nullable = false
   validation {
     condition = alltrue([
-      for k, v in var.egress_policies : contains([
+      for k, v in var.egress_policies :
+      v.from.identity_type == null || contains([
         "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
-        "ANY_USER", "ANY_SERVICE_ACCOUNT", ""
-      ], v.from.identity_type)
+        "ANY_USER", "ANY_SERVICE_ACCOUNT"
+      ], coalesce(v.from.identity_type, "-"))
```

Comment:
I think this coalesce is unnecessary. `v.from.identity_type == null` is already checked explicitly.

Reply:
@juliocc just reused the same validation than the one already existing for ingress_policies.from.identity_type, that was already supporting null usage, need to change ?

Reply:
The the coalesce is not needed in the ingress either.

@ludoo this conditions comes originally from you. Is there a reason to have the coalesce there?

Reply:
@juliocc 

did some testing, @ludoo's solution or the below test 3 solution are needed.

**test 1:** 
- with == null
- without the coalesce

result is **KO**
analysis: terraform will still evaluate the second part after ||, as coalesce is not there it is rejected

test code:
    v.from.identity_type == null || contains([
          "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
          "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT"
        ], v.from.identity_type)
terraform result:
Error: v.from.identity_type is null

** test 2:**
- without == null,
- using the same coalesce:

result is **KO**
analysis: coalesce will replace null with "-", however "-" in not in the contains list

test code:
contains([
        "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
        "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT"
      ], coalesce(v.from.identity_type, "-"))

terraform result:
    Invalid `from.identity_type` value in egress policy.

**test 3:**
- without == null
- with coalesce changed with a trick using one of the permitted value when null

result is **ok**, however the trick seems too tricky :)

test code:
contains([
        "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
        "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT"
      ], coalesce(v.from.identity_type, "IDENTITY_TYPE_UNSPECIFIED"))

terraform result:
    No changes. Your infrastructure matches the configuration.

**proposal:**
stick with ludoo's initial solution which is still tricky but less :)

    v.from.identity_type == null || contains([
          "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
          "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT"
        ], coalesce(v.from.identity_type, "-"))

Reply:
you win :)